File: RGR/app/src/main/java/com/example/rgr/Builder/Builder.java
package com.example.rgr.Builder;

import com.example.rgr.Comparator.Comparator;

import com.example.rgr.Comparator.Comparator;


public interface Builder
{
    Object createObject();
    Object readObject();
    Object clone();
    Object parseObject(String ss);
    Comparator getComparator();
    String getName();

}

File: LR_3/src/git/group/Builder/Builder.kt
package git.group.Builder

import git.group.Comparator.Comparator

interface Builder : Cloneable {
    fun createObject(): Any?
    fun readObject(): Any?
    override fun clone(): Any
    fun parseObject(ss: String?): Any?
    val comparator: Comparator?
    val name: String?
}